src/app/core/services/navigation/nav-tree.service.ts

```diff
@@ -25,50 +25,52 @@ export abstract class NavTreeService<ContentT extends RoutedContentInfo> {
   private content$ = this.currentContent.content$.pipe( // only keep those of interest for the current nav tree
     // map those which are not of interest to `undefined`
     map(content => (this.isOfContentType(content) ? content : undefined)),
-    distinctUntilChanged(), // remove multiple `undefined`
     startWith(undefined),
+    distinctUntilChanged(), // remove multiple `undefined`
     repeatLatestWhen(this.reloadTrigger),
+    switchMap(content => this.currentContent.contentHasChanged$.pipe(
+      map(() => true),
+      startWith(false),
+      map(hasChanged => ({ content, hasChanged })),
+    )),
   );
-  private children$ = merge(
-    // emit `undefined` each time the content change
-    this.content$.pipe(
-      map(c => c?.route.id),
-      distinctUntilChanged(),
-      map(() => undefined),
-    ),
-    this.content$.pipe(this.childrenNavData())
-  );
-  state$ = combineLatest([ this.children$, this.content$ ]).pipe(
-    mergeScan((prevState: FetchState<NavTreeData>, [ children, content ]) => {
+
+  state$ = this.content$.pipe(
+    switchMap(({ content, hasChanged }) => of(content).pipe(
+      this.childrenNavData(),
+      map(children => ({ children, content, hasChanged })),
+      startWith({ children: undefined, content, hasChanged })
+    )),
```

Comment:
When I check with what was written before, you clearly change the logic so that cannot be valid. 
Separating the children was not for the sake of complexifying the code, the idea was to have the children fetched only once per content (even when the same content (id) is emitted several time with a build-up).

Reply:
fixed